CONTRIBUTING.md

```diff
@@ -0,0 +1,121 @@
+
+The following guide is to meant to help you get started with contributing to Connaisseur.
+
+We are committed to positive interactions between all contributors of the project. To ensure this, please follow the [Berlin Code of Conduct](https://berlincodeofconduct.org/) in all communications.
+
+## Contents
+- [Discuss Problems, Raise Bugs and Propose Feature Ideas](#discuss-problems-raise-bugs-and-propose-feature-ideas)
+- [Contribute to Source Code](#contribute-to-source-code)
+  * [Setup the Environment](#setup-the-environment)
+  * [Test Changes](#test-changes)
+  * [Signed Commits and Pull Requests](#signed-commits-and-pull-requests)
+  * [Semantic and Conventional Commits](#semantic-and-conventional-commits)
+- [Enjoy!](#enjoy)
+
+## Discuss Problems, Raise Bugs and Propose Feature Ideas
+We are happy you made it here! If you want to raise any bugs you found, discuss issues from using Connaisseur in your own projects or have ideas for new features to be implemented, please [create an issue](https://github.com/sse-secure-systems/connaisseur/issues/new) with an informative title and description.
+
+While issues are a great way to discuss problems, bugs and new features, a direct proposal via a pull request can sometimes say more than a thousand words. So be bold and contribute to the code as described in the [next section](#contribute-to-source-code)!
+
+In case you require a more private communication, you can reach us via [connaisseur@securesystems.de](mailto:connaisseur@securesystems.de).
+
+## Contribute to Source Code
+The following steps will help you make code contributions to Connaisseur and ensure good code quality and workflow. This includes the following steps:
+1. **Setup your environment**: Setup up your local environment to best interact with the code. Further information is given in the subsection below.
+2. **Make atomic changes**: Changes should be atomic, i.e. each pull request should only fix one issue or implement one feature. As such, each pull request should ideally only contain a single commit with an concise commit message.
+3. **Test your changes**: Test any changes locally for code quality and functionality and add new tests for any additional code. How to test is described in the subsection below.
+4. **Create semantic, conventional and signed commits**: Any commits should follow a simple semantic convention to help structure the work on Connaisseur. The convention is described in the corresponding subsection. For security reasons and as the project is about signing code, we require signed commits.
+5. **Create Pull Request**: We consider code review a critical measure for code quality and security. Therefore, a pull request (PR) should be created for each contribution. It will be reviewed by one of the main contributors and potential improvements may be discussed within the PR. At least one approval is required for merging changes into the `master` branch.
```

Comment:
IMO too long and too specific. Proposal:

```
**Create Pull Request:** A pull request (PR) should be created for each contribution. It will be reviewed, and potential improvements may be discussed within the PR. After approval, changes will be merged into the `master` branch.
```

Reply:
Made the change, but would like to keep a bit of the why though obvious it shows the spirit.